articles/quickstart/webapp/rails/01-login.md

```diff
@@ -9,169 +9,124 @@ topics:
 ```
 
-Once your gems are added, install with the following command:
-
-```bash
-bundle install
-```
-
-::: note
-If you are using Windows, uncomment the `tzinfo-data` gem in the Gemfile.
-:::
+Once your gems are added, install the gems with `$ bundle install`:
 
 ### Initialize OmniAuth Auth0
 
-Add your Auth0 configuration to the `secrets.yml` file for later use:
+Create an Auth0 config file in the root of your project's directory `./config/auth0.yml` .
 
 ```yaml
+# ./config/auth0.yml
 development:
-  secret_key_base: <YOUR APPLICATION SECRET KEY>
   auth0_domain: ${account.namespace}
   auth0_client_id: ${account.clientId}
   auth0_client_secret: <YOUR AUTH0 CLIENT SECRET>
 ```
 
-:::note
-`secret_key_base` is not used for authentication but should already exist as part of your Rails app.
-:::
-
-Create a file named `auth0.rb` under `config/initializers` and configure the **OmniAuth** middleware in it.
+Create an initializer file in the root of your project's dierctory `./config/initializers/auth0.rb` and [configure](https://github.com/auth0/omniauth-auth0#additional-authentication-parameters) the **OmniAuth** middleware.
 
 ```ruby
+# ./config/initializers/auth0.rb
+Rails.application.config.auth0 = Rails.application.config_for(:auth0)
+
 Rails.application.config.middleware.use OmniAuth::Builder do
+  auth0_config = Rails.application.config_for(:auth0)
   provider(
     :auth0,
-    Rails.application.secrets.auth0_client_id,
-    Rails.application.secrets.auth0_client_secret,
-    Rails.application.secrets.auth0_domain,
+    Rails.application.config.auth0['auth0_client_id'],
+    Rails.application.config.auth0['auth0_client_secret'],
+    Rails.application.config.auth0['auth0_domain'],
     callback_path: '/auth/auth0/callback',
     authorize_params: {
-      scope: 'openid email profile'
+      scope: 'openid profile'
     }
   )
 end
 ```
 
-::: note
-This tutorial uses omniauth-auth0, a custom [OmniAuth strategy](https://github.com/intridea/omniauth#omniauth-standardized-multi-provider-authentication).
-:::
-
-### Add the Auth0 Callback Handler
-
-Use the following command to create the controller that will handle the Auth0 callback:
-
-```bash
-rails generate controller auth0 --skip-template-engine --skip-assets --no-helper
-```
-
-In the newly created controller, add success and failure callback handlers.
-
-:::note
-Rails uses a cookie by default to store session data. If you run into `CookieOverflow` error when trying to save the user info into the session, please read [the troubleshooting section](#actiondispatch-cookies-cookieoverflow) on how to store session data in memory instead.
-:::
+### Add an Auth0 controller
+Create an Auth0 controller file in the root of your project's dierctory `./config/controllers/auth0_controller.rb` for handling of the authentication callback.  Inside of the callback method assign the hash of user information returned `request.env['omniauth.auth']` to the active session.
```

Comment:
Feels like an opportunity here to include a Rails `generate` command that the reader can copy & paste to create this controller.

```
rails generate controller auth0 callback failure --no-assets --no-helper
```

WDTY? We do this later for Logout, for example.